app/assets/javascripts/admin/orders/controllers/orders_controller.js.coffee

```diff
@@ -15,12 +15,51 @@ angular.module("admin.orders").controller "ordersCtrl", ($scope, $timeout, Reque
   $scope.poll = 0
   $scope.rowStatus = {}
 
+  localStorageService.setStorageType("sessionStorage")
+  filtersInUrlKey = 'ordersFilters'
+
+  $scope.getStoredFilters = ->
+    localStorageService.get(filtersInUrlKey) || ""
+
   $scope.initialise = ->
-    $scope.per_page = 15
-    $scope.q = {
-      completed_at_not_null: true
-    }
-    $scope.fetchResults()
+    params = $location.search()
+    # Get all params from url (if presents) and then put into $scope
+    $scope.per_page = parseInt(params.per_page) || 15
+    if (params.shipping_method_id)
+      $scope.shipping_method_id = parseInt(params.shipping_method_id)
+    
+    # All params specified in the query 'q'
+    $scope.q = {}
+    searchKeys =  ["completed_at_gteq",
+     "completed_at_lteq",
+     "state_eq",
+     "number_cont",
+     "email_cont"
+     "bill_address_firstname_start",
+     "bill_address_lastname_start"
+     ]
+    for key in searchKeys when params["q["+key+"]"]
+      $scope.q[key] = params["q["+key+"]"]
+
+    # Special for arrays 'distributor_id_in' and 'order_cycle_id_in'. Arrays of arrays: q[params][]
+    for key in ["distributor_id_in", "order_cycle_id_in"] when params["q["+key+"][]"]
+      $scope.q[key] = params["q["+key+"][]"]
```

Comment:
These "special cases" look pretty wild. I imagine if we touch anything related to these filters it will be quite likely this could break.

Is there no simpler way to store and retrieve this data without having to go through every key-value pair and re-write it explicitly?

Reply:
The issues is: we have 4 different sources of truth now for the filters (the url params, the form elements, the angular variables, the localstorage) and the complexity in transferring that truth (state) between each place is getting really high.

Reply:
Yes, I understand your point. It's always painful when we are facing multiple source of truth.

I inherit from this "special cases" which were here before I try to add this feature.

Reply:
If we ignored the URL params and just focused on the variables in Angular and the data that's being stored in localStorage, could we simplify this? Like...

What happens if when we submit the form, we take the value of `$scope.q` and persist that in localStorage, then when we load the page we fetch it from localStorage and replace it in `$scope.q`?

Reply:
Why I've done both URL and localStorage:

 - Removing URL params stuff, could be great for developer experience and for futur debug/code comprehension, but I wanted also to make it clear for a user point of view. As a user, I always prefer having params in URL when I load a page to understand that filters are already filled and therefore that results are already filtered. But maybe I'm not a "basic" user. 

 - Furthermore, the aim of localStorage here is _only_ to rebuild the URL used in the "Back to order list" button:
```haml
%a.button.icon-arrow-left{icon:  'icon-arrow-left', ng: { href: admin_orders_path + "{{ '#' + getStoredFilters() || '' }}"}}
```
If we decide to trust the localStorage (and only the localStorage), we also have to invalidate it.
When we arrive to this page from another way (ie. bu clicking on the "Orders" menu that leads you to `http://localhost:3000/admin/orders`), we have to ignore localStorage (and delete it). Dealing with URL building thanks to localStorage and order_controller in the "Back to order list" is way way simpler to me

Reply:
For example, we could do something like this: https://github.com/jibees/openfoodnetwork/compare/4207-add-query-params-in-url...Matt-Yorkley:order-filters?expand=1

...then there's no need to manually mangle the data, and all the logic is simple and 100% re-usable in other contexts.

Reply:
> If we decide to trust the localStorage (and only the localStorage), we also have to invalidate it.

It gets cleared when the browser is closed, right? That's probably enough.

Reply:
> It gets cleared when the browser is closed, right? That's probably enough.

Yes it is, but I'm almost sure it's not enough. The only way to get the filter reinitialized once you go to /orders is to do it _by hand_, one by one. I don't think it's this is satisfactory

Reply:
> For example, we could do something like this:...

I will check, thanks! 🙏

Reply:
> The only way to get the filter reinitialized once you go to /orders is to do it by hand, one by one.

Yeah it really feels like it needs a button to clear all filters...

Reply:
I think it's more a question about UX / product decision no?

Reply:
> > For example, we could do something like this:...
> 
> I will check, thanks! 🙏

yes, much cleaner. I wonder if there is an opportunity to mix your solution with mine and having such a `QueryInURLPersistence` service.